examples/header-group.html

```diff
@@ -0,0 +1,80 @@
+<template id=header-group>
+
+<style>
+
+  header-group
+    { display: block }
+
+  header-group
+    hgroup
+      { background: #bada55; color: var(--darkest-color) }
+
+</style>
+
+<hgroup>
+  <h1><a href=/examples/index.html>snuggsi ツ</a></h1>
+  <slot name=header><h2>try an example</h2></slot>
+</hgroup>
+
+</template>
+
+<script>
+const tag = `header-group`
+
+Element `header-group`
+
+(class extends HTMLElement {
+
+  static onimportload (event) {
+    let template = event.target.import
+      .querySelector ('template')
+
+      document
+        .querySelector('head')
+        .appendChild (template)
+
+      // possibly should be adopting node into document
+      // and should do it on DOMContentLoaded hook up onimportload
+
+    const shadow = function (element) {
+      let fragment = template.content.cloneNode (true)
+
+      fragment.slots =
+        Array.from (fragment.querySelectorAll ('slot'))
+
+      element.slots =
+        Array.from (element.querySelectorAll ('[slot]'))
+
+      element.slots.map (function (namedslot) {
+        fragment.slots
+          .filter (slot =>
+            (slot.getAttribute ('name') === namedslot.getAttribute ('slot'))
+          )
+          .map (slot => {
+            slot.innerHTML = ''
+            , slot.appendChild (namedslot)
```

Comment:
@brandondees @mrbernnz instead of clobbering the `slot.innerHTML` then `slot.appendChild` the named slot value perhaps we should `slot.replace (namedslot)` and completely remove the `<slot>` 

Please advise.
![capture d ecran 2017-04-28 a 14 16 55](https://cloud.githubusercontent.com/assets/38223/25541682/99c5d368-2c1d-11e7-99f8-77c153d5a47d.png)